streamingccl: fix TestTenantStreamingProducerJobTimedOut flaky test

gh-casper opened the pull request:
streamingccl: fix TestTenantStreamingProducerJobTimedOut flaky test

Fixes a race between job resumer updating the job high watermark
and reading high watermark when getting a stream ingestion stats.

Closes: https://github.com/cockroachdb/cockroach/issues/84332

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/84592)
<!-- Reviewable:end -->

gh-casper committed:
streamingccl: fix TestTenantStreamingProducerJobTimedOut flaky test

Fixs a race between job resumer updating the job high watermark
and reading highwatermatk when getting a stream ingestion stats.

Release note: None

samiskin commented on pkg/ccl/streamingccl/streamingest/stream_ingestion_processor_planning.go:
```diff
@@ -194,7 +194,7 @@ func (s *streamIngestionResultWriter) AddRow(ctx context.Context, row tree.Datum
 		return errors.NewAssertionErrorWithWrappedErrf(err, `unmarshalling resolved timestamp`)
 	}
-	return s.registry.UpdateJobWithTxn(ctx, s.jobID, nil /* txn */, false, /* useReadLock */
+	return s.registry.UpdateJobWithTxn(ctx, s.jobID, nil /* txn */, true, /* useReadLock */
```
Could be good to add a TODO here around investigating why this is necessary

gh-casper commented:
bors r+
